File: runtime/tests/web_public_runtime_tests.mjs
import { resolve } from "node:path";
import { pathToFileURL } from "node:url";

const modulePath = process.argv[2];
if (!modulePath) {
  throw new Error("expected the generated Emscripten module path");
}

const { default: createMkwWebPublicRuntimeTests } = await import(
  pathToFileURL(resolve(modulePath)).href,
);
const runtime = await createMkwWebPublicRuntimeTests();
const abiVersion = runtime._MkwWebPublicRuntimeAbiVersion();
if (abiVersion !== 1) {
  throw new Error(`expected ABI version 1, got ${abiVersion}`);
}

console.log(`Web public runtime tests passed (ABI ${abiVersion}).`);

File: runtime/tests/web_public_runtime_tests.cpp
#include "platform/web_bootstrap.h"

#include <cstdio>

int main()
{
    const uint32_t actual = MkwWebPublicRuntimeAbiVersion();
    if (actual != RuntimePlatform::Web::kPublicRuntimeAbiVersion) {
        std::fprintf(stderr, "web public runtime ABI mismatch: expected %u, got %u\\n",
                     RuntimePlatform::Web::kPublicRuntimeAbiVersion, actual);
        return 1;
    }

    std::puts("web public runtime ABI test passed");
    return 0;
}
